Dockerfile

```diff
@@ -1,23 +1,11 @@
-FROM ubuntu:14.04
-MAINTAINER Erik Axelsson <erikaxelsson1@gmail.com>
+FROM node:12.16.1-buster
 
-# install our dependencies and nodejs
-RUN echo "deb http://archive.ubuntu.com/ubuntu trusty main universe" > /etc/apt/sources.list
-RUN echo "deb http://ppa.launchpad.net/chris-lea/node.js/ubuntu trusty main " > /etc/apt/sources.list
-RUN apt-get update
-RUN apt-get -y install nodejs
+LABEL maintainer="digit@chalmers.it"
 
-# use changes to package.json to force Docker not to use the cache
-# when we change our application's nodejs dependencies:
-ADD package.json /tmp/package.json
-RUN cd /tmp && npm install
-RUN mkdir -p /opt/app && cp -a /tmp/node_modules /opt/app/
+WORKDIR /usr/src/app
+COPY package*.json ./
+RUN npm ci
 
-# From here we load our application's code in, therefore the previous docker
-# "layer" thats been cached will be used if possible
-WORKDIR /opt/app
-ADD . /opt/app
+COPY . .
```

Comment:
What to this row do? Why is it needed?

Reply:
`COPY . .` Copies all of the files from the current host directory (root of project) to the containers workdirectory.
I used the offical recomendations from nodejs:  https://nodejs.org/fr/docs/guides/nodejs-docker-webapp/
tldr: We copy the package files first so that we can work on the code without needing to reinstall all dep on every docker build.

Reply:
Okay thanks :).